src/test/java/fi/otavanopisto/kuntaapi/test/server/integration/management/PostTestsIT.java

```diff
@@ -0,0 +1,171 @@
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/news/{newsArticleId}", organizationId, getOrganizationNewsArticleId(organizationId, 0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("abstract", containsString("Consectetur adipiscing elit"))
+      .body("abstract", not(containsString("more-link")));
+  }
+  
+  @Test
+  public void testListPosts() {
+    // Zeus shoud be listed before abraham and bertha because it's menu_order is set to -100 
+
+    String organizationId = getOrganizationId(0);
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/news", organizationId)
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("slug[0]", is("lorem-ipsum-dolor-sit-amet"))
+      .body("slug[1]", is("test-2"))
+      .body("slug[2]", is("test-3"));
+  } 
+  
+  @Test
+  public void testListPostsByPath() {
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/news?slug=test-3", getOrganizationId(0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(1))
+      .body("id[0]", notNullValue())
+      .body("slug[0]", is("test-3")); 
+    
+    given() 
+      .baseUri(getApiBasePath())
+      .contentType(ContentType.JSON)
+      .get("/organizations/{organizationId}/news?slug=non-existing", getOrganizationId(0))
+      .then()
+      .assertThat()
+      .statusCode(200)
+      .body("id.size()", is(0));
+  } 
+    
+  @Test
+  public void testOrganizationPostsNotFound() throws InterruptedException {
+    String organizationId = getOrganizationId(0);
+    String incorrectOrganizationId = getOrganizationId(1);
+    String organizationNewsArticleId = getOrganizationNewsArticleId(organizationId, 0);
+    
+    String[] malformedIds = new String[] {"evil", "*", "/", "1", "-1", "~"};
+    assertFound(String.format("/organizations/%s/news/%s", organizationId, organizationNewsArticleId));
+    assertEquals(3, countApiList(String.format("/organizations/%s/news", organizationId)));
+    
+    for (String malformedId : malformedIds) {
+      assertNotFound(String.format("/organizations/%s/news/%s", organizationId, malformedId));
+    }
+    
+    assertNotFound(String.format("/organizations/%s/news/%s", incorrectOrganizationId, organizationNewsArticleId));
+    assertEquals(0, countApiList(String.format("/organizations/%s/news", incorrectOrganizationId)));
+  }
+  
+  private void createPtvSettings() {
+    insertSystemSetting(PtvConsts.SYSTEM_SETTING_BASEURL, String.format("%s%s", getWireMockBasePath(), BASE_URL));
+    flushCache();
+  }
+  
+  private void deletePtvSettings() {
+    deleteSystemSetting(PtvConsts.SYSTEM_SETTING_BASEURL);
+  }
+    
+  private void createManagementSettings(String organizationId) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "createManagementSettings" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)